### PR TITLE
Rename "Todo Comments" to "Task List" to match Editor naming here.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Pulled out of https://github.com/dotnet/roslyn/pull/63902 to make that PR easier to review.

### CODE REVIEW

File: src/Features/CSharp/Portable/TaskList/CSharpTaskListService.cs [hunk 1]
```diff
@@ -11,29 +11,24 @@
...
 using Microsoft.CodeAnalysis.TodoComments;
 using Roslyn.Utilities;
 
-namespace Microsoft.CodeAnalysis.CSharp.TodoComments
+namespace Microsoft.CodeAnalysis.CSharp.TaskList
 {
-    [ExportLanguageServiceFactory(typeof(ITodoCommentService), LanguageNames.CSharp), Shared]
-    internal class CSharpTodoCommentServiceFactory : ILanguageServiceFactory
+    [ExportLanguageService(typeof(ITodoCommentService), LanguageNames.CSharp), Shared]
+    internal class CSharpTaskListService : AbstractTaskListService
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/TaskList/CSharpTaskListService.cs:21]: no need for factory.  made into only a normal service export. (same will happen for vb).

File: src/Features/LanguageServer/Protocol/Features/TaskList/TaskListListener.cs [hunk 2]
```diff
@@ -16,44 +16,45 @@
...
             CancellationToken disposalToken)
         {
             _globalOptions = globalOptions;
             _services = services;
-            _asyncListener = asynchronousOperationListenerProvider.GetListener(FeatureAttribute.TodoCommentList);
+            _asyncListener = asynchronousOperationListenerProvider.GetListener(FeatureAttribute.TaskList);
             _onTodoCommentsUpdated = onTodoCommentsUpdated;
             _disposalToken = disposalToken;
 
-            _workQueue = new AsyncBatchingWorkQueue<DocumentAndComments>(
```
Comment (CyrusNajmabadi) [inline src/Features/LanguageServer/Protocol/Features/TaskList/TaskListListener.cs:56]: removed helper type that really wasn't serving a useful purpose.

File: src/Features/LanguageServer/Protocol/Features/TaskList/TaskListOptionsStorage.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.TaskList;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Features.TaskList
+{
+    internal static class TaskListOptionsStorage
```
Comment (CyrusNajmabadi) [inline src/Features/LanguageServer/Protocol/Features/TaskList/TaskListOptionsStorage.cs:12]: rename, but not picked up by github unfortunately.

File: src/VisualStudio/Core/Test/Diagnostics/TodoListTableDataSourceTests.vb [hunk 4]
```diff
@@ -26,13 +25,13 @@ Namespace Microsoft.VisualStudio.LanguageServices.UnitTests.Diagnostics
                 Dim provider = New TestTodoListProvider()
```
Comment (dibarbet) [inline src/VisualStudio/Core/Test/Diagnostics/TodoListTableDataSourceTests.vb:25]: nit - rename file

**Review by dibarbet — APPROVED** [review]

File: src/VisualStudio/Core/Def/TableDataSource/TaskList/VisualStudioTodoListTable.cs [hunk 5]
```diff
@@ -5,37 +5,37 @@
...
 using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.TaskList;
 using Microsoft.VisualStudio.Shell.TableManager;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.TableDataSource
 {
-    [ExportEventListener(WellKnownEventListeners.TodoListProvider, WorkspaceKind.Host), Shared]
-    internal class VisualStudioTodoListTableWorkspaceEventListener : IEventListener<ITodoListProvider>
+    [ExportEventListener(WellKnownEventListeners.TaskListProvider, WorkspaceKind.Host), Shared]
+    internal class VisualStudioTaskListTableWorkspaceEventListener : IEventListener<ITaskListProvider>
```
Comment (dibarbet) [inline src/VisualStudio/Core/Def/TableDataSource/TaskList/VisualStudioTodoListTable.cs:17]: nit -rename file

**Pull request merged**